packages/queued-request-controller/tests/QueuedRequestMiddleware.test.ts

```diff
@@ -0,0 +1,425 @@
+import type { AddApprovalRequest } from '@metamask/approval-controller';
+import { ControllerMessenger } from '@metamask/base-controller';
+import type {
+  NetworkControllerGetNetworkClientByIdAction,
+  NetworkControllerGetStateAction,
+  NetworkControllerSetActiveNetworkAction,
+  NetworkControllerSetProviderTypeAction,
+} from '@metamask/network-controller';
+import { serializeError } from '@metamask/rpc-errors';
+import type { SelectedNetworkControllerSetNetworkClientIdForDomainAction } from '@metamask/selected-network-controller';
+import { SelectedNetworkControllerActionTypes } from '@metamask/selected-network-controller';
+
+import type { QueuedRequestControllerEnqueueRequestAction } from '../src/QueuedRequestController';
+import { createQueuedRequestMiddleware } from '../src/QueuedRequestMiddleware';
+
+const buildMessenger = () => {
+  return new ControllerMessenger<
+    | SelectedNetworkControllerSetNetworkClientIdForDomainAction
+    | QueuedRequestControllerEnqueueRequestAction
+    | NetworkControllerGetStateAction
+    | NetworkControllerSetActiveNetworkAction
+    | NetworkControllerSetProviderTypeAction
+    | NetworkControllerGetNetworkClientByIdAction
+    | AddApprovalRequest,
+    never
+  >();
+};
+
+const buildMocks = (messenger: any, mocks: any = {}) => {
```

Comment:
RE: @mcmire https://github.com/MetaMask/core/pull/1806/files#r1364156133
The typing's a little involved, but should be accurate.

```suggestion
const buildMocks = (
  messenger: ReturnType<typeof buildMessenger>,
  mocks = {} as Partial<
    Record<
      | ((
          | NetworkControllerActions
          | ApprovalControllerActions
          | QueuedRequestControllerActions
        )['type'] extends `${
          | 'Network'
          | 'Approval'
          | 'QueuedRequest'}Controller:${infer T}`
          ? T
          : never)
      | keyof typeof SelectedNetworkControllerActionTypes,
      jest.Mock
    >
  >,
) => {
```

Reply:
ah I didnt see this, but thank you for that